ts/main.ts

```diff
@@ -83,44 +143,54 @@ function swapView(anime: Anime): void {
   const img = document.createElement('img');
   img.src = anime.images.jpg.image_url;
   img.alt = anime.title;
-
-  // Clear existing children in imageElement and append new image
-  while (imageElement.firstChild) {
-    imageElement.removeChild(imageElement.firstChild);
-  }
   imageElement.appendChild(img);
 
   // Update rating
   ratingElement.textContent = `Rating: ${anime.rating ?? 'N/A'}`;
 
+  // Add or Remove from Favorites button
+  const isFavorite = favorites.some((fav) => fav.mal_id === anime.mal_id);
+  const starButton = createStarButton(isFavorite);
+  starButton.addEventListener('click', () => {
+    updateFavorites(anime);
+    const isFavoriteNow = favorites.some((fav) => fav.mal_id === anime.mal_id);
+    starButton
+      .querySelector('path')!
+      .setAttribute('fill', isFavoriteNow ? 'yellow' : 'none');
+  });
```

Comment:
Ideally this would be available all of the time instead in the details view and you find a way to determine what anime is currently listed on the event listener. It looks like this is in your `viewSwap` function and it just adds on swap?